sklearn/metrics/_classification.py

```diff
@@ -883,7 +912,28 @@ class labels [2]_.
     n_classes = confusion.shape[0]
     sum0 = np.sum(confusion, axis=0)
     sum1 = np.sum(confusion, axis=1)
-    expected = np.outer(sum0, sum1) / np.sum(sum0)
+
+    mgs_changing_default = (
+        "The default return value of `cohen_kappa_score` in case of a division "
+        "by zero has been deprecated in 1.7 and will be changed to 0.0 in version "
+        "1.9. Set `replace_undefined_by=0.0` to use the new default and to silence "
+        "this Warning."
+    )
+
+    numerator = np.outer(sum0, sum1)
+    denominator = np.sum(sum0)
+    if np.isclose(denominator, 0):
+        if replace_undefined_by == "deprecated":
+            replace_undefined_by = np.nan
+            warnings.warn(mgs_changing_default, FutureWarning)
+        msg = (
+            "`y2` contains no labels that are presented in both `y1` and `labels`."
+            "cohen_kappa_score is undefined and set to the value defined in "
+            "the `replace_undefined_by` param, which defaults to `np.nan`."
+        )
+        warnings.warn(msg, UndefinedMetricWarning, stacklevel=2)
+        return replace_undefined_by
```

Comment:
This part can be in a function defined here (inside `kohen_kappa_score`) to avoid duplicating the logic twice here.